crates/query-engine/translation/src/translation/query/native_queries.rs

```diff
@@ -38,6 +38,16 @@ pub fn translate(env: &Env, state: State) -> Result<Vec<sql::ast::CommonTableExp
                             models::Argument::Variable { name } => match &variables_table {
                                 Err(err) => Err(err.clone()),
                                 Ok(variables_table) => Ok(values::translate_variable(
+                                    // We need a 'State' value when translating variables in order
+                                    // to be able to generate fresh names for bound relational
+                                    // expressions.
+                                    // However, we cannot readily re-use the 'state' argument we're
+                                    // given, since that is an "owned" value which is "moved" in
+                                    // the course of iterating over the native queries accumulated
+                                    // within.
+                                    // Ideally we should resolve this by tracking native queries
```

Comment:
Not quite "just" pass it by reference.

Passing by reference results in a borrowing conflict, since we are looping over the native queries (also contained within the `state` value). Both the loop and `translate_variables` want to borrow the reference to the `state` value.

(From my understanding this prohibits `translate_variables` from mutating the list of native queries which the loop is iterating over, which makes sense IMO, as it _would_ be able to do just that with a `&mut State` reference.)

It's true that we could also clone our way out of this, but this also seems wasteful to me, and doesn't address the issue that we'd be passing somewhat nonsensical data to `translate_variables`, in the shape of redundant native queries that (according to the types) look like you could insert new ones into, but which would just be ignored.